### PR TITLE
Add DockerHub release workflow

### PR DESCRIPTION
**Pull request by p1ngod**
As stated in https://github.com/jboss-dockerfiles/wildfly/issues/145 adds GitHub Actions workflow to build and publish tags to DockerHub repo.

Please create a CLI/token login for DockerHub and two corresponding GitHub secrets in advance:
- **DOCKERHUB_USERNAME**
- **DOCKERHUB_TOKEN**

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: publish-release
+on:
+  push:
+    tags:
+      - "*.Final"
+jobs:
+  docker:
+    runs-on: ubuntu-latest
```
Comment (jmesnil) [inline .github/workflows/publish-release.yml:8]: could you please add a `environment: dockerhub`?

```
jobs:
  docker:
    environment: dockerhub
    runs: ...
```

File: .github/workflows/publish-release.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        uses: docker/metadata-action@v3
+        with:
+          images: jboss/wildfly
+          tags: type=ref,event=tag
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
```
Comment (jmesnil) [inline .github/workflows/publish-release.yml:23]: could you please rename it to `${{ secrets.USERNAME }}`

File: .github/workflows/publish-release.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
```
Comment (jmesnil) [inline .github/workflows/publish-release.yml:24]: could you please rename it to `${{ secrets.TOKEN }}`?

**Review by jmesnil — CHANGES_REQUESTED** [review]
Your PR works fine as it is but I'd like to use a `dockerhub` environment to manage the job and its secrets.

@goldmann I've already created the `dockerhub` environment as well as the `USERNAME` secret in it (set to `jboss`).
Could you please generate a token for https://hub.docker.com/r/jboss/wildfly and add it to a `TOKEN` secret in the `dockerhub` environment?

File: .github/workflows/publish-release.yml (hunk 3 above)
Comment (p1ngod) [reply .github/workflows/publish-release.yml]: done

File: .github/workflows/publish-release.yml (hunk 2 above)
Comment (p1ngod) [reply .github/workflows/publish-release.yml]: done

**Comment by jmesnil** [comment]
@p1ngod as discussed on wildfly-dev mailing list, I opened #151 (based on your commit) to build and push images to a quay.io image repository

**Comment by p1ngod** [comment]
So as there will be no future dockerhub build, I will close this PR

**Pull request closed without merging**